## What is cointegration?
The idea behind cointegration is that even if the price of two different assets both follow random walk, it is still possible that a linear regression combination of them is not a random walk.

In [1]:
import pandas as pd

import warnings
warnings.filterwarnings("ignore")

In [2]:
data=pd.read_csv('datasets/2015-01-01 to 2021-05-13.csv')

data_kbl_gbime=data[data['Stock Symbol'].isin(['KBL','GBIME'])]

data_kbl_gbime['Date']=pd.to_datetime(data_kbl_gbime.Date)
kbl=data_kbl_gbime[data_kbl_gbime['Stock Symbol']=='KBL']
gbime=data_kbl_gbime[data_kbl_gbime['Stock Symbol']=='GBIME']



gbime_kbl_merge=kbl.merge(gbime,on="Date",how="inner")

kbl=gbime_kbl_merge[gbime_kbl_merge['Stock Symbol_x']=='KBL']
gbime=gbime_kbl_merge[gbime_kbl_merge['Stock Symbol_y']=='GBIME']

In [3]:
# Import the statsmodels module for regression and the adfuller function
import statsmodels.api as sm
from statsmodels.tsa.stattools import adfuller

# Regressing GBIME on KBL
kbl = sm.add_constant(kbl['Closing Price_x'])
result = sm.OLS(gbime['Closing Price_y'],kbl['Closing Price_x']).fit()

# Computing ADF
#print(result.summary())
b = result.params[0]
adf_stats = adfuller(gbime['Closing Price_y'] - b*kbl['Closing Price_x'])
print("The p-value for the ADF test is ", adf_stats[1])

The p-value for the ADF test is  0.12213377297573313


####  The data shows that the KBL and GBIME are not cointegrated as the probabily that the null HYpothesis that the KBL and GBIME are not-cointegrated is 12%. which is signigicantly high probability.That why we cant reject the null hypothesis.